src/plugins/dashboard/public/application/utils/use/use_chrome_visibility.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Any modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+import { useState, useEffect } from 'react';
```

Comment:
I think this is exactly the same as in visualize. I am okay with adding a file for feature branch to unblock the later development. Maybe we could open an issue to have a discussion to create a package for osd hooks or make it to a single place.